src/main/java/emissary/test/core/ExtractionTest.java

```diff
@@ -256,7 +257,27 @@ protected void checkAnswers(Element el, IBaseDataObject payload, List<IBaseDataO
             byte viewData[] = payload.getAlternateView(viewName);
             assertTrue(String.format("Alternate View '%s' is present, but should not be, in %s", viewName, tname), viewData == null);
         }
+    }
 
+    public void checkAttachmentAndExtractedRecordCount(Element el, IBaseDataObject payload, List<IBaseDataObject> attachments, String tname)
+            throws DataConversionException {
+        int numAtt = JDOMUtil.getChildIntValue(el, "numAttachments");
```

Comment:
If I don't define numAttachments, then I can still have att elements in the xml and the test do not fail.

Reply:
Yep, after I stopped working last night I thought of a lot of these cases. Will push a new commit today to look out for these use cases.